### PR TITLE
Temporary removing string slice span bench test.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Work around while we wait for:
https://github.com/dotnet/coreclr/pull/10544

### CODE REVIEW

**Comment by ahsonkhan** [comment]
cc @AndyAyersMS, @jkotas.

I have to break the circular dependency here.

**Comment by AndyAyersMS** [comment]
LGTM.

File: tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs [hunk 1]
```diff
@@ -1108,43 +1108,7 @@ public static void TestSpanNonPortableCastFromIntToByte(int length)
         }
         #endregion
 
-        #region TestSpanSliceStringChar<T>
```
Comment (jkotas) [inline tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs]: It would be better to just to put this under `#if false`

**Comment by jkotas** [comment]
LGTM otherwise

File: tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs (hunk 1 above)
Comment (ahsonkhan) [inline tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs]: I agree, but I plan to add this back fairly quickly.

**Comment by ahsonkhan** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Pull request merged**